src/gui_common/CustomDialog.cs

```diff
@@ -206,25 +207,19 @@ public override void _Notification(int what)
             case NotificationResized:
             {
                 UpdateChildRects();
-
-                if (FullRect)
-                    SetToFullRect();
-
+                ApplyRectSettings();
                 break;
             }
 
             case NotificationVisibilityChanged:
             {
                 if (Visible)
                 {
-                    OnShown();
-
-                    if (FullRect)
-                        SetToFullRect();
+                    ApplyRectSettings();
                 }
                 else
                 {
-                    OnHidden();
```

Comment:
If we want that unified hide signal that's emitted for both modal and non-modal then there's `hide()` or `visibility_changed`, I think that is what my reasoning was when changing this if I remember correctly.

Reply:
Didn't we do it so that the hidden and popup hidden were forwarded from our custom signal? That made it work for both cases. Otherwise *all* of the dialog code everywhere need to take into account the two different ways, which is why I highly recommend we keep that custom unified close signal.

Reply:
Readded.